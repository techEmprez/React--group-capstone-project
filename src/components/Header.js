import React from 'react'

 const Header = () => {
  return (
      <div>Header
      <img className='navLogo' src= {Logo} alt="NavLogo" />
    </div>
  )
}

export default Header;
