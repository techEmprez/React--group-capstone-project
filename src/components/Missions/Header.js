/* eslint-disable react/no-unescaped-entities */
import React from 'react';
import NavLink from './NavLink';
import Logo from '../../assets/capstone.png';
import '../../App.css';

const Header = () => (
  <div className="header">
    <img className="navLogo" src={Logo} alt="NavLogo" />
    <h1 className="pageTitle">Space Travelers' Hub</h1>
    <NavLink />
  </div>
);

export default Header;
