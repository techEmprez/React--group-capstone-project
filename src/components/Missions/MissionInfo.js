/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable no-console */
/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect } from 'react';
import uuid from 'react-uuid';
import { useSelector, useDispatch } from 'react-redux';
import '../../App.css';
import { fetchMissionsList, joinMission } from '../../redux/missions/mission';

const MissionInfo = (props) => {
  const { missionsList } = useSelector((state) => state.allMissions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissionsList());
  }, []);

  console.log(missionsList);

  const handleJoinMission = (event) => {
    event.preventDefault();
    dispatch(joinMission(event.target.parentElement.parentElement.id));
    console.log('Mission Joined');
  };

  return (
    <div className="mission-container">
      { missionsList.map((mission) => (
        <table key={uuid()} className="missionDataTable">
          <tbody className="dataBody">
            <tr className="missionDataRow" id={mission.mission_id}>
              <td className="mission1Name">{mission.mission_name}</td>
              <td className="mission1Desc">{mission.description}</td>
              <td className="mission1Status">
                { mission.isMissionJoined
                  ? <div className="status1">{props.joinedStatus}</div>
                  : <div className="status2">{props.notJoinedStatus}</div>}
              </td>
              <td className="mission1Btn">
                { mission.isMissionJoined
                  ? <button type="button" className="leaveBtn" onClick={handleJoinMission}>Leave Mission</button>
                  : <button type="button" className="joinBtn" onClick={handleJoinMission}>Join Mission</button>}
              </td>
            </tr>
          </tbody>
        </table>
      ))}
    </div>

  );
};

MissionInfo.defaultProps = {
  notJoinedStatus: 'NOT A MEMBER',
  joinedStatus: 'ACTIVE MEMBER',
};

export default MissionInfo;
