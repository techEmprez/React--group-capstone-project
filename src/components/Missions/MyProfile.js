/* eslint-disable no-console */
import React from 'react';
import { useSelector } from 'react-redux';
import { ListGroup, Container } from 'react-bootstrap';

const MyProfile = () => {
  const joinedMission = useSelector((state) => state.allMissions.missionsList);
  // eslint-disable-next-line max-len
  const filterJoinedMissions = joinedMission.filter((joinedMissionList) => joinedMissionList.isMissionJoined);
  console.log(joinedMission);

  const rockets = useSelector((state) => state.rocket);
  const filteredRockets = rockets.filter((rocket) => rocket.reserved === true);

  return (
    <Container className="d-flex justify-content-center mt-5">
      <ListGroup className="w-25 me-5">
        <h2>My Missions</h2>
        {filterJoinedMissions.length === 0 ? (
          <p className="text-danger">Please would you like to join the mission with David?</p>
        ) : (
          filterJoinedMissions.map((mission) => (
            <ListGroup.Item key={mission.mission_id}>{mission.mission_name}</ListGroup.Item>
          ))
        )}
      </ListGroup>
      <ListGroup className="w-25">
        <h2>My Rockets</h2>
        {filteredRockets.length === 0 ? (<p className="text-danger">Hey! Kindly pick a rocket.</p>
        ) : (
          filteredRockets.map((rocket) => (
            <ListGroup.Item key={rocket.id}>{rocket.name}</ListGroup.Item>
          ))
        )}
      </ListGroup>
    </Container>
  );
};

export default MyProfile;
