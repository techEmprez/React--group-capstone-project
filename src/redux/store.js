import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketSlice';

export default configureStore({
  reducer: {
    rocket: rocketReducer,
  },
});
