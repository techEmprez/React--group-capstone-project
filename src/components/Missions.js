/* eslint-disable no-console */
/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect } from 'react';
import uuid from 'react-uuid';
import { useSelector, useDispatch } from 'react-redux';
import MissionInfo from './MissionInfo';
// import MissionsData from './MissionsData';
import { fetchMissionsList } from '../redux/missions/mission';

const Missions = () => {
  const { missionsList } = useSelector((state) => state.allMissions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissionsList());
  }, []);

  console.log(missionsList);

  const renderMissionsList = missionsList.map((mission) => (
    <MissionInfo
      key={uuid()}
      name={mission.mission_name}
      description={mission.description}
      status={mission.status}
    />
  ));

  return (
    <div>
      {renderMissionsList}
    </div>
  );
};

export default Missions;
