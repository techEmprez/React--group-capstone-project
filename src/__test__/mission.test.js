import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Missions from '../components/Missions';
import Header from '../components/Header';
import store from '../redux/configureStore';

test('Unit Test Check For Mission Component', () => {
  render(<Provider store={store}><Missions /></Provider>);
  const tableElement = screen.getByText(/Missions/);
  expect(tableElement).toBeInTheDocument();
});

describe('Test Header Component', () => {
  it('Test Header Component renders correctly', () => {
    const component = TestRenderer.create(
      <Router>
        <Header />
      </Router>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Test Missions component', () => {
  it('Test Mission Component Renders Correctly', () => {
    const component = TestRenderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});

test('Unit Test Check For Mission Component', () => {
  render(<Provider store={store}><Missions /></Provider>);
  const tableElement = screen.getByText(/Description/);
  expect(tableElement).toBeInTheDocument();
});

test('Unit Test Check For Mission Component', () => {
  render(<Provider store={store}><Missions /></Provider>);
  const tableElement = screen.getByText(/Status/);
  expect(tableElement).toBeInTheDocument();
});
