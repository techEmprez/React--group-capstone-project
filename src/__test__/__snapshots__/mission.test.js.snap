// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header Component Test Header Component renders correctly 1`] = `
<div
  className="header"
>
  <img
    alt="NavLogo"
    className="navLogo"
    src="capstone.png"
  />
  <h1
    className="pageTitle"
  >
    Space Travelers' Hub
  </h1>
  <nav>
    <ul
      className="navBar"
    >
      <li
        className="navItem"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="navItem"
      >
        <a
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="navLine"
      >
        |
      </li>
      <li
        className="navItem"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Test Missions component Test Mission Component Renders Correctly 1`] = `
<div>
  <hr />
  <table
    className="mission-table-head"
  >
    <tbody>
      <tr
        className="table-head"
      >
        <th
          className="missionName"
        >
          Missions
        </th>
        <th
          className="missionDesc"
        >
          Description
        </th>
        <th
          className="missionStatus"
        >
          Status
        </th>
      </tr>
    </tbody>
  </table>
  <div
    className="mission-container"
  />
</div>
`;
